app/Listeners/TaskUpdateMessage.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Events\TaskUpdate;
+use App\GenericModel;
+use App\Profile;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class TaskUpdateMessage
+{
+
+    protected $tasks;
+
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  TaskUpdate $event
+     * @return void
+     */
+    public function handle(TaskUpdate $event)
+    {
+        $this->tasks = $event->tasks;
+
+        //task user id
+        $user_id = $this->tasks->task_history[0]['user'];
+
+        $task_xp = $this->tasks->xp;
+
+        $profile = Profile::find($user_id);
+
+        //avoid milliseconds
+        $submitted = $this->tasks->due_date - intval($this->tasks->task_history[0]['timestamp'] / 1000);
```

Comment:
Instead of hardcoding `0` index for last record, iterate through all history records and check for biggest timestamp - biggest one is for qa passed